pennylane/pauli/pauli_arithmetic.py

```diff
@@ -404,11 +443,39 @@ def __mul__(self, other):
 
         for pw1 in self:
             for pw2 in other:
-                prod_pw, coeff = pw1 * pw2
+                prod_pw, coeff = pw1 @ pw2
                 final_ps[prod_pw] = final_ps[prod_pw] + coeff * self[pw1] * other[pw2]
 
         return final_ps
 
+    def __mul__(self, other):
+        """Multiply a PauliSentence by a scalar"""
+        if isinstance(other, PauliSentence):
+            # this is legacy support and will be removed after a deprecation cycle
+            return self @ other
+
+        if isinstance(other, TensorLike):
+            if not qml.math.ndim(other) == 0:
+                raise ValueError(
+                    f"Attempting to multiply a PauliSentence with an array of dimension {qml.math.ndim(other)}"
+                )
+
+            return PauliSentence({key: other * value for key, value in self.items()})
+
+        raise TypeError(
+            f"PauliSentence can only be multiplied by numerical data. Attempting to multiply by {other} of type {type(other)}"
+        )
+
+    __rmul__ = __mul__
```

Comment:
Same comment as above about either fully overriding or leaving this be till later.